pkg/server/cluster_server.go

```diff
@@ -129,6 +140,17 @@ func (cs *clusterServer) GetConfig(cr poolRequest) (*runtime.RawExtension, error
 		return nil, fmt.Errorf("could not get controllerconfig: %w", err)
 	}
 
+	// we cannot mock this in a test env
+	if cs.configMapLister != nil {
+		cm, err := cs.configMapLister.ConfigMaps(ctrlcommon.MCONamespace).Get("kubeconfig-data")
+		if err != nil {
+			klog.Errorf("Could not get kubeconfig data: %v", err)
```

Comment:
I did some testing and it definitely seems that the workers come up too fast for the operator to have made this available.

It seems fine afterwards though for node scaling purposes